mantidproject/mantid
# Muon workspace grouping

**Description of work.**

The muon GUI was slow and this was mainly due to the overzealous calls to group workspace. This changes the code so it only calls group workspace at the end of the calculations. I have preserved the output in the ADS from previous versions. 

I have also added a small speed up in the plotting by changing the way autoscale works.

**To test speed ups**

On the main branch (release build) open muon analysis (MA) and load EMU 114150-60 and time how long it takes (on my machine its about 16 sec)

Get this branch and turn on the model fitting (see https://stfc365.sharepoint.com/:w:/r/sites/Mantid/_layouts/15/Doc.aspx?sourcedoc=%7B908F6EE2-0342-4843-90BE-7217122A9F08%7D&file=Feature%20flag%20instructions.docx&action=default&mobileredirect=true)

Open mantid
Open MA
load EMU 114150-60 and time how long it takes. This should be a lot faster than main.

Turn off autoscale and zoom in on the plot
Turn autoscale back on and check that works


**To test grouping is still correct**

Open MA
Load some data (e.g. MUSR 62260-1)
Check that all of the workspaces are in the correct groups (if you load N runs you should have N groups)

Change Rebin to fixed
Enter 10
Check the new workspaces have been added to the correct groups

Go to the grouping tab
Click the `+` below one of the tables and enter a name (e.g. test)
Check that the new workspaces are in the correct groups

Go to the phase table tab
Click `calculate phase table` and call it "table"
Check its added to the correct group (should only be in the group that matches the workspace on the first row of the table)
Click the `+` at the bottom and give it a  name (e.g. phase)
Check that the workspaces are added to the correct groups

Go to fitting
Add a function, click fit
You should get a new group with 3 workspaces (data, parameters and covarience matrix) 
Tick the simultaneous box, change the fit function (add a flat background) and do another fit
Another group should appear (for the suggested runs you will have 6 things in the group)

Go to sequential fit
Press `sequentially fit all`
Check you have the new group

Go to results and click `Output Results`
A table and a group (will include the word combinations) will appear in the ADS

Go to the Model fitting tab
Add a function and press fit
A new group will appear in the ADS

Close MA, clear the ADS

Open frequency domain analysis
Load MUSR 62263
Go to the transform tab
Click `calculate`
The workspaces should be added to the group
Change the drop down at the top of the tab from FFT to maxent
Tick all of the checkboxes in the top table
Click calculate - this is slow
When its done all of the workspaces should be added to the group


Fixes #32860. 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
